app/src/org/commcare/mediadownload/MissingMediaDownloadHelper.kt

```diff
@@ -10,27 +10,29 @@ import org.commcare.engine.resource.AndroidResourceUtils
 import org.commcare.engine.resource.AppInstallStatus
 import org.commcare.engine.resource.ResourceInstallUtils
 import org.commcare.engine.resource.installers.LocalStorageUnavailableException
-import org.commcare.preferences.HiddenPreferences
-import org.commcare.resources.model.*
 import org.commcare.network.RequestStats
+import org.commcare.preferences.HiddenPreferences
 import org.commcare.resources.ResourceInstallContext
+import org.commcare.resources.model.*
 import org.commcare.tasks.ResultAndError
-import org.commcare.util.LogTypes
 import org.commcare.utils.AndroidCommCarePlatform
 import org.commcare.utils.FileUtil
 import org.commcare.utils.StringUtils
 import org.commcare.views.dialogs.PinnedNotificationWithProgress
 import org.commcare.views.notifications.NotificationMessage
 import org.commcare.views.notifications.NotificationMessageFactory
 import org.javarosa.core.services.Logger
-import java.lang.Exception
 import java.util.concurrent.TimeUnit
 
 // Contains helper functions to download lazy or missing media resources
 object MissingMediaDownloadHelper : TableStateListener {
 
+    @Volatile
     private var mResourceInProgressListener: MissingMediaDownloadListener? = null
+
+    @Volatile
     private var resourceInProgress: Resource? = null
```

Comment:
any vars which are shared between thread should be marked as `volatile` to ensure that the writes to these are published to all threads immediately